apps/sap-commerce-cloud/frontend/src/api/fetchProductList.ts

```diff
@@ -46,3 +37,93 @@ export async function fetchProductList(
   }
   return { products: [], errors: responseJson['errors'] };
 }
+
+type FetchHAAProductListParams = {
+  url: string;
+  baseSite: string;
+  searchQuery: string;
+  parameters: SAPParameters;
+  updateTotalPages: UpdateTotalPagesFn;
+  ids: BaseAppSDK['ids'];
+  cma: CMAClient;
+};
+const fetchHAAProductList = async ({
+  baseSite,
+  searchQuery,
+  url,
+  parameters,
+  updateTotalPages,
+  ids,
+  cma,
+}: FetchHAAProductListParams): Promise<ProductListResponse> => {
+  const { response } = await cma.appActionCall.createWithResponse(
+    {
+      appActionId: 'fetchProductList',
+      environmentId: ids.environment,
+      spaceId: ids.space,
+      appDefinitionId: ids.app!,
+    },
+    {
+      parameters: {
+        sapApiEndpoint: url,
+        apiKey: ids.app,
+      },
+    }
+  );
+  const responseJson = JSON.parse(response.body);
```

Comment:
I'll need to verify that this is the correct shape returned from the HAA.
Also might need to handle response body not JSON errors